modules/augment/src/averageblur.cpp

```diff
@@ -5,41 +5,41 @@
 #include "precomp.hpp"
 namespace cv { namespace augment {
 
-AverageBlur::AverageBlur(Size _minKernelSize, Size _maxKernelSize)
+AverageBlur::AverageBlur(Size _minKernelSize, Size _maxKernelSize, const std::vector<int>& _borderTypes)
 {
+    if (_borderTypes.size() > 0) borderTypes = _borderTypes;
+
+    else borderTypes = { BORDER_CONSTANT , BORDER_REPLICATE , BORDER_REFLECT};
+
     minKernelSize = _minKernelSize;
     maxKernelSize = _maxKernelSize;
+    kernelSize = _minKernelSize;
     sameXY = false;
 }
 
-AverageBlur::AverageBlur(int _minKernelSize, int _maxKernelSize)
-{
-    minKernelSize = Size(_minKernelSize, _minKernelSize);
-    maxKernelSize = Size(_maxKernelSize, _maxKernelSize);
-    sameXY = true;
-}
-
-AverageBlur::AverageBlur(int _kernelSize)
-{
-    minKernelSize = maxKernelSize = Size(_kernelSize, _kernelSize);
-    sameXY = true;
-}
-
-AverageBlur::AverageBlur(Size _kernelSize)
+AverageBlur::AverageBlur(Size _kernelSize, int _borderType)
 {
+    borderTypes = { _borderType };
     minKernelSize = maxKernelSize = _kernelSize;
     sameXY = false;
 }
 
 void AverageBlur::init(const Mat&)
 {
-    kernelSize.height = Transform::rng.uniform(minKernelSize.height / 2, maxKernelSize.height / 2 + maxKernelSize.height % 2) * 2 + 1; //generate only random odd numbers
-    kernelSize.width = sameXY? kernelSize.height : Transform::rng.uniform(minKernelSize.width / 2, maxKernelSize.width / 2 + maxKernelSize.width % 2) * 2 + 1; //generate only random odd numbers
+    if (minKernelSize == maxKernelSize) kernelSize = minKernelSize;
```

Comment:
If `minKernelSize == maxKernelSize`, `kernelSize` equals `minKernelSize` (see constructor)

Reply:
what do you mean?

Reply:
I mean that every iteration this code assign `minKernelSize` to `kernelSize` however we already did it once in the constructor:
```cpp
    minKernelSize = _minKernelSize;
    maxKernelSize = _maxKernelSize;
    kernelSize = _minKernelSize;
    sameXY = false;
}
```

BTW, please check that it's initialized in all the constructors.

Reply:
Yes I see

Reply:
done